### ISSUE TITLE
Bug: [consistent-type-assertions] false positive "allow-as-parameter" and TaggedTemplate

### ISSUE THREAD

**Issue by yeonjuan** [typescript-eslint/typescript-eslint#8866]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.3&fileType=.tsx&code=C4TwDgpgBAKlC8UCGA7EBuAUAEwgYwBskAnaPAexQGdgoAjEgLmTSweIAoBvKAM3PLMAjFAC%2ByKrACUbEgAMAJFx79BUEeKSSYouViA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWgENkyRNHyUuqDAG1w2HFOgdokADTKVWFdkjjJ02UwDKRJOijjtug5A4AjAFaIy%2BADKV%2B0UfAAKiSIAIISUjJyNkbw8BwA7mLItMSi-gC2iH6QDhAAvg4AusqF%2BUA&tsconfig=&tokens=false

### Repro Code

```TypeScript
type T = any;
declare const bar: any;
bar({ foo: 1 } as T);
bar`${{ foo: 1 } as T}`;
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/consistent-type-assertions": [
      "error",
      {
        "assertionStyle": "as",
        "objectLiteralTypeAssertions": "allow-as-parameter"
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I expected that line 4 should not report an error.

### Actual Result

There was an error on the line 4
